### PR TITLE
AOB-2

### PR DESCRIPTION
**Pull request by solivier**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | MOB reviewed
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/Attribute/AuthorFilter.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+use Pim\Component\Catalog\Query\Filter\FieldFilterInterface;
+
+/**
+ * Author filter or an Elasticsearch query.
+ *
+ * @author    Olivier Soulet <olivier.soulet@akeneo.com>
+ * @copyright 2018 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class AuthorFilter extends AbstractFieldFilter implements FieldFilterInterface
```
Comment (momoss) [inline src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/Attribute/AuthorFilter.php]: Put this Filter in EE

File: src/Pim/Bundle/CatalogBundle/Resources/config/query_builders.yml [hunk 2]
```diff
@@ -59,6 +59,16 @@ services:
             - '@pim_catalog.factory.product_cursor'
             - '@pim_catalog.query.product_query_builder_resolver'
 
+    pim_catalog.query.product_proposal_query_builder_factory:
```
Comment (momoss) [inline src/Pim/Bundle/CatalogBundle/Resources/config/query_builders.yml]: Put the service definition in EE

File: src/Pim/Bundle/DataGridBundle/Datasource/ProductDatasource.php [hunk 3]
```diff
@@ -52,7 +52,7 @@ public function getResults()
         $context = [
             'locales'             => [$this->getConfiguration('locale_code')],
             'channels'            => [$this->getConfiguration('scope_code')],
-            'data_locale'         => $this->getParameters()['dataLocale'],
+            'data_locale'         => $this->getConfiguration('locale_code'),
```
Comment (momoss) [inline src/Pim/Bundle/DataGridBundle/Datasource/ProductDatasource.php]: Revert

File: src/Pim/Bundle/DataGridBundle/Datasource/ProductDatasource.php [hunk 4]
```diff
@@ -109,7 +109,7 @@ private function normalizeEntityWithValues(EntityWithValuesInterface $item, arra
     {
         $defaultNormalizedItem = [
             'id'               => $item->getId(),
-            'dataLocale'       => $this->getParameters()['dataLocale'],
+            'dataLocale'       => $this->getConfiguration('locale_code'),
```
Comment (momoss) [inline src/Pim/Bundle/DataGridBundle/Datasource/ProductDatasource.php]: Revert

File: src/Pim/Bundle/DataGridBundle/Resources/config/pagers.yml [hunk 5]
```diff
@@ -18,3 +18,4 @@ services:
                 - 'product-group-grid'
                 - 'published-product-grid'
                 - 'association-product-picker-grid'
+                - 'proposal-grid'
```
Comment (momoss) [inline src/Pim/Bundle/DataGridBundle/Resources/config/pagers.yml]: EE

File: src/Pim/Bundle/InstallerBundle/Command/DatabaseCommand.php [hunk 6]
```diff
@@ -126,6 +126,7 @@ protected function resetElasticsearchIndex(OutputInterface $output)
     {
         $output->writeln('<info>Reset elasticsearch indexes</info>');
 
+        $this->getContainer()->get('akeneo_elasticsearch.client.product_proposal')->resetIndex();
```
Comment (momoss) [inline src/Pim/Bundle/InstallerBundle/Command/DatabaseCommand.php]: EE

File: src/Akeneo/Bundle/ElasticsearchBundle/tests/integration/ResetIndexesCommandIntegration.php [hunk 7]
```diff
@@ -37,6 +37,13 @@ private function assertIndexesNotEmpty(): void
         $esClients = $this->get('akeneo_elasticsearch.registry.clients')->getClients();
 
         foreach ($esClients as $esClient) {
+            if ($this->hasParameter('product_proposal_index_name')) {
```
Comment (skeleton) [inline src/Akeneo/Bundle/ElasticsearchBundle/tests/integration/ResetIndexesCommandIntegration.php]: ouch, on CE??

File: src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/IdentifierFilter.php [hunk 8]
```diff
@@ -88,9 +88,9 @@ public function addAttributeFilter(
     protected function checkValue($property, $operator, $value)
     {
         if (Operators::IN_LIST === $operator || Operators::NOT_IN_LIST === $operator) {
-            FieldFilterHelper::checkArray($property, $value, self::class);
+            FieldFilterHelper::checkArray($property, $value, static::class);
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/IdentifierFilter.php]: I think it's useless now as we created our own filters

Comment (momoss) [reply src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/IdentifierFilter.php]: Exact no need static at this place

**Comment by momoss** [comment]
[Last build one broken behat](https://ci.akeneo.com/blue/organizations/jenkins/akeneo%2Fpim-community-dev/detail/PR-7435/54/pipeline/)

**Pull request merged**